frontend/lib/src/components/widgets/DataFrame/DataFrame.tsx

```diff
@@ -233,13 +213,18 @@ function DataFrame({
 
   const { columns: originalColumns } = useColumnLoader(element, data, disabled)
 
-  // On the first rendering, try to load initial widget state if
-  // it exist. This is required in the case that other elements
-  // are inserted before this widget.
+  /**
+   * On the first rendering, try to load the initial editing state
+   * from widget state if it exists. This is required in the case
+   * that other elements are inserted before this widget.
```

Comment:
Could you elaborate a little bit how other elements can impact the editing state of a dataframe widget?

Reply:
Added more to the comment. TLDR: if elements are inserted before, it can happen that the components gets unmounted and loses all state.

Reply:
btw. this is the case (problem) for all widgets and even non-widgets e.g. tabs or expander can lose the state as well because of this